x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/InferenceBaseRestTest.java

```diff
@@ -0,0 +1,127 @@
+        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).build();
+    }
+
+    static String mockServiceModelConfig() {
+        return """
+            {
+              "service": "test_service",
+              "service_settings": {
+                "model": "my_model",
+                "api_key": "abc64"
+              },
+              "task_settings": {
+                "temperature": 3
+              }
+            }
+            """;
+    }
+
+    protected Map<String, Object> putModel(String modelId, String modelConfig, TaskType taskType) throws IOException {
+        String endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("PUT", endpoint);
+        request.setJsonEntity(modelConfig);
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    protected Map<String, Object> getModels(String modelId, TaskType taskType) throws IOException {
+        var endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("GET", endpoint);
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    protected Map<String, Object> inferOnMockService(String modelId, TaskType taskType, List<String> input) throws IOException {
+        var endpoint = Strings.format("_inference/%s/%s", taskType, modelId);
+        var request = new Request("POST", endpoint);
+
+        var bodyBuilder = new StringBuilder("{\"input\": [");
+        for (var in : input) {
+            bodyBuilder.append('"').append(in).append('"').append(',');
+        }
+        // remove last comma
+        bodyBuilder.deleteCharAt(bodyBuilder.length() - 1);
+        bodyBuilder.append("]}");
+
+        request.setJsonEntity(bodyBuilder.toString());
+        var reponse = client().performRequest(request);
+        assertOkWithErrorMessage(reponse);
+        return entityAsMap(reponse);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void assertNonEmptyInferenceResults(Map<String, Object> resultMap, TaskType taskType) {
+        if (taskType == TaskType.SPARSE_EMBEDDING) {
+            var results = (List<String>) resultMap.get("result");
+            assertThat(results, not(empty()));
+            for (String result : results) {
+                assertThat(result, is(not(emptyString())));
+            }
+        } else {
+            fail("test with task type [" + taskType + "] are not supported yet");
+        }
+    }
+
+    protected static void assertOkWithErrorMessage(Response response) throws IOException {
+        int statusCode = response.getStatusLine().getStatusCode();
+        if (statusCode == 200 || statusCode == 201) {
+            return;
+        }
```

Comment:
The next line is `String responseStr = EntityUtils.toString(response.getEntity());`. The entity can only be read once, after it is converted to a string the it cannot be used again. The purpose is to avoid reading the entity so it can be used later if the assertion will not trip.